src/main/java/com/exadel/discountwebapp/category/service/CategoryService.java

```diff
@@ -41,12 +43,10 @@ public CategoryResponseVO create(CategoryRequestVO request) {
 
     @Transactional(propagation = Propagation.REQUIRED)
     public CategoryResponseVO update(Long id, CategoryRequestVO request) {
-        Category category = categoryRepository.findById(id).orElse(null);
-        if (category != null) {
-            categoryMapper.updateEntity(request, category);
-            return categoryMapper.toVO(categoryRepository.save(category));
-        }
-        return null;
+        Category category = categoryRepository.findById(id).
+                orElseThrow(() -> new EntityNotFoundException("Could not find category with id: " + id));
+        categoryMapper.updateEntity(request, category);
+        return categoryMapper.toVO(categoryRepository.save(category));
```

Comment:
it is better to perform separate operations in different lines

categoryRepository.save(category)
categoryMapper.toVO(...

Reply:
done